src/NexusMods.App.UI/Controls/Navigation/NavigationControl.cs

```diff
@@ -0,0 +1,118 @@
+using System.Diagnostics;
+using System.Reactive;
+using Avalonia;
+using Avalonia.Controls;
+using Avalonia.Input;
+using JetBrains.Annotations;
+using NexusMods.App.UI.WorkspaceSystem;
+using ReactiveUI;
+
+namespace NexusMods.App.UI.Controls.Navigation;
+
+[PublicAPI]
+public class NavigationControl : Button
+{
+    public static readonly StyledProperty<ReactiveCommand<NavigationInput, Unit>?> NavigateCommandProperty =
+        AvaloniaProperty.Register<NavigationControl, ReactiveCommand<NavigationInput, Unit>?>(nameof(NavigationCommand));
+
+    public ReactiveCommand<NavigationInformation, Unit>? NavigationCommand
+    {
+        get
+        {
+            var value = GetValue(CommandProperty);
+            return value as ReactiveCommand<NavigationInformation, Unit>;
+        }
+        set => SetValue(CommandProperty, value);
+    }
+
+    private NavigationInformation NavigationInformation
+    {
+        get => (NavigationInformation)CommandParameter!;
+        set => CommandParameter = value;
+    }
+
+    private readonly ReactiveCommand<OpenPageBehaviorType, Unit> _contextMenuCommand;
+
+    public NavigationControl()
+    {
+        _contextMenuCommand = ReactiveCommand.Create<OpenPageBehaviorType>(openPageBehaviorType =>
+        {
+            NavigationInformation = NavigationInformation.From(openPageBehaviorType);
+            OnClick();
+        });
+
+        var contextMenu = new ContextMenu
+        {
+            Items =
+            {
+                new MenuItem
+                {
+                    Header = "Open in new tab",
+                    Command = _contextMenuCommand,
+                    CommandParameter = OpenPageBehaviorType.NewTab,
+                },
+                new MenuItem
+                {
+                    Header = "Open in new panel",
+                    Command = _contextMenuCommand,
+                    CommandParameter = OpenPageBehaviorType.NewPanel,
+                },
```

Comment:
I think we can should make this option disabled if you have MaxPanels already

Reply:
That would be the most user-friendly, but I'd take too long to add. We can do this later, this PR is already big enough.

Reply:
Ok, please open an issue or make a note in an open one for this